x/mongo/driver/operation.go

```diff
@@ -602,6 +604,13 @@ func (op Operation) Execute(ctx context.Context) error {
 			return err
 		}
 
+		// If a query interceptor is present, allow it to check the operation and short circuit with an error
+		if op.CommandInterceptor != nil {
```

Comment:
I think we'd likely want to have this higher up in the `Execute` function, ideally before it does any work like checking out a connection. I just put it here since it was a place where the wire message had already been built.